src/pytom_tm/matching.py

```diff
@@ -262,6 +297,51 @@ def run(self) -> tuple[npt.NDArray[float], npt.NDArray[float], dict]:
                 square_sum_kernel(self.plan.ccc_map * roi_mask) / roi_size
             )
 
+            if self.noise_correction:
+                # Rotate noise template texture into template
+                self.plan.random_phase_template_texture.transform(
+                    rotation=(rotation[0], rotation[1], rotation[2]),
+                    rotation_order='rzxz',
+                    output=self.plan.template,
+                    rotation_units='rad'
+                )
+
+                if self.plan.wedge is not None:
+                    # Add wedge to the template after rotating
+                    self.plan.template = irfftn(
+                        rfftn(self.plan.template) * self.plan.wedge,
+                        s=self.plan.template.shape
+                    )
+
+                # Normalize and mask template
+                mean = mean_under_mask(self.plan.template, self.plan.mask, mask_weight=self.plan.mask_weight)
+                std = std_under_mask(self.plan.template, self.plan.mask, mean, mask_weight=self.plan.mask_weight)
+                self.plan.template = ((self.plan.template - mean) / std) * self.plan.mask
+
+                # Paste in center
+                self.plan.template_padded[pad_index] = self.plan.template
+
+                # Fast local correlation function between volume and template, norm is the standard deviation at each
+                # point in the volume in the masked area
+                self.plan.ccc_map = (
+                        irfftn(self.plan.volume_rft_conj * rfftn(self.plan.template_padded),
+                               s=self.plan.template_padded.shape)
+                        / self.plan.std_volume
+                )
```

Comment:
this part is very similar to 257-285 and should in my opinion be split out into a separate function

Reply:
Did it!